### PR TITLE
Fetch gene panel data by molecular profile ids

### PR DESCRIPTION
**Pull request by adamabeshouse**
Fixes https://github.com/cBioPortal/cbioportal/issues/8603

### CODE REVIEW

File: src/pages/resultsView/ResultsViewPageStore.ts [hunk 1]
```diff
@@ -2034,7 +2029,7 @@ export class ResultsViewPageStore {
         },
     });
 
-    readonly coverageInformation = remoteData<CoverageInformation>(
+    /*readonly coverageInformation = remoteData<CoverageInformation>(
```
Comment (adamabeshouse) [inline src/pages/resultsView/ResultsViewPageStore.ts:2032]: oops I forgot to delete this

File: src/shared/lib/GenePanelUtils.ts [hunk 2]
```diff
@@ -143,58 +144,52 @@ export function computeGenePanelInformation(
...
-                    molecularProfileId: profile.molecularProfileId,
-                    sampleId,
-                });
-            }
-        }
-    });
-    // query for gene panel data using sample molecular identifiers
-    let genePanelData: GenePanelData[];
-    if (sampleMolecularIdentifiers.length && genes.result!.length) {
+    // fetch all gene panel data for profiles
```
Comment (alisman) [inline src/shared/lib/GenePanelUtils.ts:152]: amazing commentry (commentary?)
thank you

Comment (alisman) [reply src/shared/lib/GenePanelUtils.ts:152]: commenting

**Comment by alisman** [comment]
@adamabeshouse now that we fetch by profile, i believe we should be able to fire the fetch earlier in the "timeline" because we no longer need to wait for samples.    this could potentially shave off seconds from load time of page.  could you investigate if this is true?    we would pass this data into getCoverageInformation

File: src/pages/resultsView/ResultsViewPageStore.ts [hunk 3]
```diff
@@ -2053,7 +2048,22 @@ export class ResultsViewPageStore {
...
+    readonly coverageInformation = remoteData<CoverageInformation>({
+        await: () => [
+            this.molecularProfilesInStudies,
+            this.sampleKeyToSample,
+            this.patients,
+            this.genes,
+        ],
+        invoke: () =>
+            getCoverageInformation(
+                this.molecularProfilesInStudies.result!,
```
Comment (kalletlak) [inline src/pages/resultsView/ResultsViewPageStore.ts:2061]: Do we need gene-panel data for all molecular profiles in studies or just the selected profiles?

Comment (adamabeshouse) [reply src/pages/resultsView/ResultsViewPageStore.ts:2061]: good catch, yeah I think you're right

**Comment by adamabeshouse** [comment]
@kalletlak @alisman updated

**Review by kalletlak — APPROVED** [review]

**Pull request merged**